### PR TITLE
API updates and fixes

### PR DESCRIPTION
**Pull request by yathishram**
### Description

- This PR contains changes to the APIs for client interaction

### Modified

- `connectUser` takes `apiKey` and `secret` for connecting to threadDB
- `getLoginUser` returns `null` if there is no user instead of an `error`
-  Encoding of files happens on `TextEncoder`
-  Maintains the `connection` object on the SDK rather than passing them every time

### Added

- `queryUser` - API for querying and searching users using their email

### Removed

-  Removes `beneficiary` argument from `createNewSafe`
- `TextEncoder` removed from `helpers.js`

### CODE REVIEW

File: src/SafientCore.ts [hunk 1]
```diff
@@ -379,8 +409,14 @@ export class SafientCore {
       if(result[0].onChain === true && result[0].stage === safeStages.ACTIVE){
 
         evidenceUri = await this.utils.createClaimEvidenceUri(file, evidenceName, description)
-        tx = await this.claims.safientMain.createClaim(result[0]._id, evidenceUri)
-        txReceipt = await tx.wait()
+        if(result[0].claimType === ClaimType.ARBITRATION){
+          tx = await this.claims.safientMain.createClaim(result[0]._id, evidenceUri)
+          txReceipt = await tx.wait()
+        }else{
```
Comment (koshikraj) [inline src/SafientCore.ts:415]: Add `else if` instead.

File: src/types/types.ts [hunk 2]
```diff
@@ -45,7 +45,9 @@ export type SafeData = {
   stage: number;
   encSafeKeyShards: Shard[];
   claims: Claims[];
-  onChain: boolean
+  onChain: boolean;
+  claimType: number;
```
Comment (koshikraj) [inline src/types/types.ts:49]: We should import the ClaimType from `@safient/claims` and use it here.

File: src/SafientCore.ts [hunk 3]
```diff
@@ -15,6 +15,12 @@ import {SafientClaims} from "@safient/claims"
 import {ethers} from "ethers"
 require('dotenv').config();
 
+
+const ClaimType = {
```
[the remark this answers is not in the document]
Comment (yathishram) [reply src/SafientCore.ts:19]: We can use that as well. I'll try doing that

[the remark this answers is not in the document]
Comment (koshikraj) [reply src/SafientCore.ts:19]: You need to update the `safient-contract-js` as well.

File: test/scenario4.js [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    const loginUser = await guardianThreeSc.safientCore.getLoginUser(guardianThree.idx.id);
+    expect(loginUser.name).to.equal('Guardian 3');
+    expect(loginUser.email).to.equal('guardianThree@test.com');
+});
+
+  //should create a safe onChain and offChain
+  it('Should create safe with "Testing Safe data" with Signal Based Claim', async () => {
+   
+     
+      safeId = await creatorSc.safientCore.createNewSafe(creator.idx.id, beneficiary.idx.id, "Testing safe Data", true, 0, 6)
```
Comment (koshikraj) [inline test/scenario4.js:135]: Pass the claim type instead of `int`. You can import from `@safient/claims`

File: test/scenario4.js [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+    const file = {
+        name: "signature.jpg"
+    }
+    disputeId = await beneficiarySc.safientCore.claimSafe(safeId, file, "Testing Evidence", "Lorsem Text")
+    expect(disputeId).to.be.a('number');
+  });
+
+  it('Should send signal after claim', async () => {
+
+      const result = await creatorSc.safientCore.sendSignal(safeId) //Passing a claim
```
Comment (koshikraj) [inline test/scenario4.js:152]: This is actually failing the claim.

File: test/scenario4.js [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+
+  it('Should create a claim', async () => {
+    const file = {
+        name: "signature.jpg"
+    }
+    disputeId = await beneficiarySc.safientCore.claimSafe(safeId, file, "Testing Evidence", "Lorsem Text")
+    expect(disputeId).to.be.a('number');
+  });
+
+it('Should wait for somtime', async () => {
```
Comment (koshikraj) [inline test/scenario4.js:164]: Don't make this is a separate test. You can add the delay in the previous test itself.

File: package.json [hunk 7]
```diff
@@ -36,7 +36,7 @@
     "@magic-ext/oauth": "^0.7.0",
```
Comment (koshikraj) [inline package.json:36]: Update the package version  to `0.1.1-alpha`

**Review by koshikraj — CHANGES_REQUESTED** [review]

**Review by koshikraj — APPROVED** [review]

**Pull request merged**